File: dotnet/RailsLib.Net/Net/Common/GameData.cs
using Newtonsoft.Json;
using System.Collections.Generic;


namespace GameLib.Net.Common
{
    [JsonObject(MemberSerialization.Fields)]
    public class GameData
    {
        private GameInfo game;
    private GameOptionsSet gameOptions;
    private List<string> players;

        public GameData() { }

        private GameData(GameInfo game, GameOptionsSet gameOptions, List<string> players)
        {
            this.game = game;
            this.gameOptions = gameOptions;
            this.players = players;
        }

        public static GameData Create(GameInfo game, GameOptionsSet.Builder gameOptions, List<string> players)
        {
            return new GameData(game, gameOptions.Build(players.Count), players);
        }

        public string GameName
        {
            get
            {
                return game.Name;
            }
        }

        public GameOptionsSet GameOptions
        {
            get
            {
                return gameOptions;
            }
        }

        public List<string> Players
        {
            get
            {
                return players;
            }
        }
    }
}

File: src/main/java/net/sf/rails/common/GameData.java
package net.sf.rails.common;

import java.util.ArrayList;
import java.util.List;

public class GameData {
    private final GameInfo game;

    private final GameOptionsSet gameOptions;

    private final List<String> players;

    private GameData(GameInfo game, GameOptionsSet gameOptions, List<String> players) {
        super();

        this.game = game;
        this.gameOptions = gameOptions;
        this.players = new ArrayList<>(players);
    }

    public static GameData create(GameInfo game, GameOptionsSet.Builder gameOptions, List<String> players) {
        return new GameData(game, gameOptions.withNumberOfPlayers(players.size()).build(), players);
    }

    public String getGameName() {
        return game.getName();
    }

    public GameOptionsSet getGameOptions() {
        return gameOptions;
    }

    public List<String> getPlayers() {
        return players;
    }
}
